Source/mxm/GB_colscale.c

```diff
@@ -217,6 +228,26 @@ GrB_Info GB_colscale                // C = A*D, column scale with diagonal D
         }
         #endif
 
+        // We are using the CPU. Finish the dup from A -> C.
+        if (info == GrB_NO_VALUE)
+        {
+            // copy A->p, A->h into C->p, C->h
+            size_t psize = A->p_is_32 ?
+                sizeof (uint32_t) : sizeof (uint64_t) ;
+            size_t isize = A->i_is_32 ?
+                sizeof (uint32_t) : sizeof (uint64_t) ;
+            int64_t anvec = A->nvec ;
+            int nthreads_max = GB_Context_nthreads_max ( ) ;
+
+            if (A->p != NULL)
+            {
+                GB_memcpy (C->p, A->p, (anvec+1) * psize, nthreads_max) ;
```

Comment:
We need to assert that C->p and A->p have the same integer type.  Likewise C->h and A->h.

Reply:
Fixed - this is already known to be true from the results of `GB_dup_worker`, and we use the same assumption elsewhere, e.g. in the kernels. Should the same assert be there too?